### PR TITLE
Webhooks overall cleanup

### PR DESCRIPTION
**Pull request by piotrci**
Goal: improve the quality and consistency of docs related to webhooks.

I have gone through the files and improved the language. I found a number of inconsistencies (info missing from /beta or v1.0) and fixed them.

### CODE REVIEW

File: api-reference/beta/resources/webhooks.md [hunk 1]
```diff
@@ -175,17 +193,19 @@ When the user receives an email, Microsoft Graph sends a notification like the f
 }
 ```
 
-Note the value object contains a list. If there are many queued notifications, Microsoft Graph sends them in a single request.
+Note the `value` field is an array of objects. When there are many queued notifications, Microsoft Graph may send multiple items in a single request.
 
 ## Processing the notification
 
-After your application starts receiving notifications, it must process them. The following are the minimum tasks that your app must perform to process a notification:
+Each notification received by your app should be processed; the following are the minimum tasks that your app must perform to process a notification:
```
Comment (dkershaw10) [inline api-reference/beta/resources/webhooks.md]: minimum tasks -> minimum set of tasks

File: api-reference/v1.0/resources/webhooks.md [hunk 2]
```diff
@@ -13,29 +13,29 @@ Using the Microsoft Graph REST API, an app can subscribe to changes on the follo
...
 
-Or a specific ID:
-`users/{id}`, `groups/{id}`, `groups/{id}/conversations`
-
 Or to a top-level resource:
 `me/messages`, `me/contacts`, `me/events`, `users`, or `groups`
 
-Or on a Sharepoint / OneDrive for Business drive:
+Or to a specific object:
+`users/{id}`, `groups/{id}`, `groups('{id}')/conversations`
```
Comment (dkershaw10) [inline api-reference/v1.0/resources/webhooks.md]: Odd that we don't have consistent info here.  There's nothing wrong with `groups('{id}')/conversations` - i.e. it works, but in the same line we have "key as segment" which is the way we normally doc the API.

Would recommend changing to `groups/{id}/conversations`

**Comment by dkershaw10** [comment]
@piotrci Minor changes required. Otherwise LGTM!

File: api-reference/beta/resources/webhooks.md [hunk 3]
```diff
@@ -15,29 +15,29 @@ Using the Microsoft Graph REST API, an app can subscribe to changes on the follo
...
 
-For instance, you can create a subscription to a specific folder:
+For instance, you can create a subscription to a specific mail folder:
 `me/mailfolders('inbox')/messages`
 
 Or to a top-level resource:
 `me/messages`, `me/contacts`, `me/events`, `users`, or `groups`
 
-Or to a specific id:
+Or to a specific object:
```
[the remark this answers is not in the document]
Comment (piotrci) [reply api-reference/beta/resources/webhooks.md]: Resource instance? we seem to use "resource" for the class of objects (messages, events, users are "resources"). Here we addressing a specific resource item? what would be a good term?

File: api-reference/beta/resources/webhooks.md [hunk 4]
```diff
@@ -15,29 +15,29 @@ Using the Microsoft Graph REST API, an app can subscribe to changes on the follo
...
-Or on a user's personal OneDrive:
+Or to a user's personal OneDrive:
 `/drives/{id}/root`
 `/drives/{id}/root/subfolder`
 
-After Microsoft Graph accepts the subscription request, it pushes notifications to the URL specified in the subscription. The app then takes action according to its business logic. For example, it fetches more data, updates cache and views, etc.
+After Microsoft Graph accepts the subscription request, it pushes notifications to the URL specified in the subscription. The app then takes action according to its business logic. For example, it fetches more data, updates its cache and views, etc.
 
-Apps need to renew their subscriptions before the expiration time. Otherwise, they need to create a new subscription. For a list of maximum expiration times, see [Maximum length of subscription per resource type](subscription.md#maximum-length-of-subscription-per-resource-type).
+Apps need to renew their subscriptions before they expire, otherwise, they will need to create a new subscription. For a list of maximum expiration times, see [Maximum length of subscription per resource type](subscription.md#maximum-length-of-subscription-per-resource-type).
```
[the remark this answers is not in the document]
Comment (piotrci) [reply api-reference/beta/resources/webhooks.md]: Yes - i did check on a random internet page about punctuation, so - science! :)

**Comment by lleonard-msft** [comment]
#hold-off 

@piotrci - Please resolve the merge conflicts.

File: api-reference/beta/resources/webhooks.md [hunk 5]
```diff
@@ -63,46 +63,64 @@ Creating a subscription is the first step to start receiving notifications for a
...
+## Notification endpoint validation
+
+Microsoft Graph validates the notification endpoint provided in the `notificationUrl` property of the subscription request before creating the subscription. The validation process occurs as follows:
 
 1. Microsoft Graph sends a POST request to the notification URL:
 
   ```
   POST https://{notificationUrl}?validationToken={TokenDefinedByMicrosoftGraph}
-  clientState: {Data sent in ClientState value in subscription request (if any)}
+  clientState: {data originally provided in the clientState field of the subscription creation request (if any)}
```
Comment (jasonjoh) [inline api-reference/beta/resources/webhooks.md]: I don't see that clientState is sent in validation requests.

**Comment by jasonjoh** [comment]
per conversation with Peter, merging his webhooks PR's in to the webhooks branch with my existing webhook PR's so we can reconcile any conflicts and have one central PR to review.

**Pull request merged**